File: Dice Combinations.py
#tle for some reason
MOD = 10**9+7

def countWays(n):
    if n == 0:
        return 1
    dp = [0]*(n+1)
    dp[0] = 1
    
    for i in range(1, n+1):
        for j in range(1, 7):
            if j<=i:
                dp[i] = (dp[i] + dp[i-j]) % MOD
    
    return dp[n]

print(countWays(int(input())))

File: Dice Combinations.cpp
#include <bits/stdc++.h>

using namespace std;

const int MOD = 1000000007;

int countWays(int n) {
    if (n == 0) {
        return 1;
    }
    vector<int> dp(n + 1, 0);
    dp[0] = 1;

    for (int i = 1; i <= n; ++i) {
        for (int j = 1; j <= 6; ++j) {
            if (j <= i) {
                dp[i] = (dp[i] + dp[i - j]) % MOD;
            }
        }
    }

    return dp[n];
}

int main() {
    int n;
    cin >> n;
    cout << countWays(n) << endl;
    return 0;
}
